modules/Movecraft/src/main/java/net/countercraft/movecraft/listener/InteractListener.java

```diff
@@ -99,27 +122,26 @@ public void onPlayerInteractStick(PlayerInteractEvent event) {
                 int DY = 1;
                 if (event.getPlayer().isSneaking())
                     DY = -1;
-
+                if (craft.getType().getGearShiftsAffectDirectMovement())
+                    DY *= currentGear;
                 craft.translate(0, DY, 0);
                 timeMap.put(event.getPlayer(), System.currentTimeMillis());
                 craft.setLastCruiseUpdate(System.currentTimeMillis());
                 return;
             }
             // Player is onboard craft and right clicking
-            float rotation = (float) Math.PI * event.getPlayer().getLocation().getYaw() / 180f;
-
-            float nx = -(float) Math.sin(rotation);
-            float nz = (float) Math.cos(rotation);
-
-            int dx = (Math.abs(nx) >= 0.5 ? 1 : 0) * (int) Math.signum(nx);
-            int dz = (Math.abs(nz) > 0.5 ? 1 : 0) * (int) Math.signum(nz);
-            int dy;
 
+            final Vector direction = player.getLocation().getDirection();
```

Comment:
This may be causing problems as you can no longer go up and diagonally at the same time.